### PR TITLE
Bugfix/crash on morph relations

### PR DESCRIPTION
**Pull request by gael-connan-cybex**
### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Added conditional checks to prevent crashes when dealing with polymorph relations in higher Laravel 9 versions.
- Refactored the code to provide less information about polymorph relations, ensuring compatibility and stability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ModelReflector.php</strong><dd><code>Fix crash and enhance handling of polymorph relations</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/ModelReflector.php

<li>Added conditional checks for polymorph relations.<br> <li> Prevented crash by handling methods that may not exist.<br> <li> Refactored code to provide less information for certain relations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/laravel-reflector/pull/12/files#diff-face89c2dc57e01fac4eb6fe94a18b30b42ea36ab196b24dfe0c835097de5c53">+16/-10</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by gael-connan-cybex** [comment]
All (implemented) Retaylor buttons have been tested locally with this patch. The crashes around user management have demonstrably stopped occuring.

File: src/ModelReflector.php [hunk 1]
```diff
@@ -97,20 +97,26 @@ public function getModelRelations($model, bool $forceRefresh = false): Collectio
 
         foreach ($reflectionMethods as $reflectionMethod) {
             if ($this->reflectionMethodIsRelation($reflectionMethod)) {
-                $methodName   = $reflectionMethod->getName();
-                $relation     = $modelInstance->{$methodName}();
-                $relatedModel = $relation->getRelated();
-
+                $methodName                    = $reflectionMethod->getName();
```
Comment (mszulik) [inline src/ModelReflector.php:100]: Why are there so many spaces? Should probably reformat once more.

Comment (gael-connan-cybex) [reply src/ModelReflector.php]: yes

**Review by mszulik — APPROVED** [review]
seems to work

**Review by jheusinger — APPROVED** [review]
doesnt crash in the other app anymore

**Pull request merged**